internal/config/config.go

```diff
@@ -285,8 +285,9 @@ type BaseOperatorConf struct {
 	FilterChildAnnotationPrefixes []string `default:""`
 	// adds compare-options and sync-options for prometheus objects converted by operator.
 	// It helps to properly use converter with ArgoCD
-	PrometheusConverterAddArgoCDIgnoreAnnotations bool `default:"false"`
-	EnabledPrometheusConverterOwnerReferences     bool `default:"false"`
+	PrometheusConverterAddArgoCDIgnoreAnnotations bool   `default:"false"`
+	EnabledPrometheusConverterOwnerReferences     bool   `default:"false"`
+	VMServiceScrapeDefaultRole                    string `default:"endpoints"`
```

Comment:
I think var `VMServiceScrapeEnforeEndpointslices` might be easier to validate, wdyt?

Reply:
then it will be hard to reuse the same for other possible values of discoveryRole. Want to keep it like this to reuse for all possible values of discoveryRole.

Reply:
`VMServiceScrapeEnforeEndpointslices` can only work when `VMServiceScrapeSpec.DiscoveryRole` is not specified.
So maybe rename to `VMServiceScrapeDefaultRoleEndpointslices`:)

Reply:
Expected behaviour for `VMEnforceEndpointslices` option:
1) changes default value of discoveryRole from `endpoints` to `endpointslices` during vmagent config build https://github.com/VictoriaMetrics/operator/blob/master/internal/controller/operator/factory/vmagent/servicescrape.go#L28
2) keep behavior the same for explicitly set value of `discoveryRole`


 It solves issue with `endpoints` not only for objects created from `ServiceMonitor`, but also for `VMServiceScrape` components.